### PR TITLE
Model recipe

### PR DESCRIPTION
**Pull request by aravadikesh**
Introduce preliminary code for a BERT recipe aimed at classification tasks.

Implements TextClassificationDataset for efficient tokenization and batching of text and label data.

BERTTextClassifier Class:
Training Pipeline: Loads data, encodes labels, initializes model/tokenizer, and trains using HuggingFace's Trainer API.
Evaluation: Computes accuracy, precision, recall, and F1-score using scikit-learn .
Model Saving: Persists model, tokenizer, label encoder, and metadata (including hyperparameters and training results).
Inference: Loads trained models and performs batch predictions with optional probability outputs.
Metadata Handling: Stores and loads all relevant training and model information for reproducibility.
Logging: Integrates a configurable logging setup for better traceability.

### CODE REVIEW

File: src/mc_classifier_pipeline/bert_recipe.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+            raise ValueError(f"Required columns '{text_column}' and '{label_column}' not found in training data")
+
+        return train_df, test_df
+
+    def prepare_model(self, num_labels: int):
+        """Initialize tokenizer and model"""
+        logger.info(f"Loading tokenizer and model: {self.model_name}")
+
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name)
+        self.model = AutoModelForSequenceClassification.from_pretrained(self.model_name, num_labels=num_labels)
```
Comment (Celaena24) [inline src/mc_classifier_pipeline/bert_recipe.py:100]: Can you add problem type to this. I think it was throwing an error because sometimes AutoModelForSequenceClassification was incorrectly auto-detecting the problem type and using MSE loss (regression) instead of CrossEntropy loss (classification).

`
self.model = AutoModelForSequenceClassification.from_pretrained(
            self.model_name, num_labels=num_labels, problem_type="single_label_classification"
        )
`

**Review by Celaena24 — COMMENTED** [review]
Overall good looks great!
